setup-env

```diff
@@ -50,6 +59,14 @@ while (("$#")); do
       INSTALL_HOOKS=1
       shift
       ;;
+    -v | --version)
+      PYTHON_VERSION=$2
```

Comment:
You should check that a version is actually specified on the command line.  This would involve:
- Checking that there is actually a second argument.
- Possibly also checking that the second argument matches a regex that represents a python version.

Reply:
Re the first bullet, right now the script has this (undesirable) behavior:
```
$ ./setup-env --version
./setup-env: line 64: $2: unbound variable
```

Instead it should give a nice error message.

Reply:
The first bullet is taken care of by `getopt`.  @michaelsaki, if you can create an issue for the second bullet then I can resolve this conversation.

Reply:
@jsf9k So right now I have some conditional checks for valid installed versions of Python. So if anyone enters anything that isn't listed when `pyenv versions --bare --skip-aliases --skip-envs` is ran then it will say that it isn't installed and exits the script. I could make a few more changes so it checks specifically for a `version.major.minor` if you think that would be a better look. But I wanted to make sure to point out first. See here https://github.com/cisagov/skeleton-generic/blob/1240bdd98fe3523dada222782a55df8c2b258e5c/setup-env#L105-L114 and also here https://github.com/cisagov/skeleton-generic/blob/1240bdd98fe3523dada222782a55df8c2b258e5c/setup-env#L180-L187

Reply:
Yes, I think the code should also be checking to make sure that the version specified is a semantic version, since that's the versioning scheme that Python follows.  This is a future improvement that I just want to capture in an issue before it is lost in the sands of time.

Actually, I'll just go ahead and create the issue and resolve this conversation.

Reply:
See cisagov/skeleton-generic#167.